lib/iris/coords.py

```diff
@@ -684,6 +766,13 @@ def _as_defn(self):
                          self.units, self.attributes, self.coord_system)
         return defn
 
+    # Must supply __hash__ as Python 3 does not enable it if __eq__ is defined.
+    # NOTE: Violates "objects which compare equal must have the same hash".
+    # Currently needed, but not really correct and should be fixed.
```

Comment:
I'd add "for merging" to this comment because otherwise it might not be clear _why_ it is needed (that is, what's using it).